src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs

```diff
@@ -0,0 +1,229 @@
+            }
+
+            var command = string.Join(" ", new[] {scriptCommand}.Concat(scriptArguments.Skip(1)));
+            return command;
+        }
+
+        internal string ReplaceScriptVariables(string command)
+        {
+            foreach (var scriptVariableEntry in ScriptVariableToMSBuildMap)
+            {
+                var scriptVariableName = scriptVariableEntry.Key;
+                var msbuildMapping = scriptVariableEntry.Value;
+
+                if (command.Contains($"%{scriptVariableName}%"))
+                {
+                    if (msbuildMapping == null)
+                    {
+                        throw new Exception(
+                            $"{scriptVariableName} is currently an unsupported script variable for project migration");
+                    }
+
+                    command = command.Replace($"%{scriptVariableName}%", msbuildMapping);
+                }
+            }
+
+            return command;
+        }
+
+        private bool IsPathRootedForAnyOS(string path)
+        {
+            return path.StartsWith("/") || path.Substring(1).StartsWith(":\\");
+        }
+
+        private ProjectTargetElement CreateTarget(ProjectRootElement csproj, string scriptSetName)
+        {
+            var targetName = $"{scriptSetName[0].ToString().ToUpper()}{string.Concat(scriptSetName.Skip(1))}Script";
+            var targetHookInfo = ScriptSetToMSBuildHookTargetMap[scriptSetName];
+
+            var target = csproj.AddTarget(targetName);
+            if (targetHookInfo.IsRunBefore)
+            {
+                target.BeforeTargets = targetHookInfo.TargetName;
+            }
+            else
+            {
+                target.AfterTargets = targetHookInfo.TargetName;
+            }
+
+            return target;
+        }
+
+        private void AddExec(ProjectTargetElement target, string command)
+        {
+            var task = target.AddTask("Exec");
+            task.SetParameter("Command", command);
+        }
+
+        // ProjectJson Script Set Name to 
+        private static Dictionary<string, TargetHookInfo> ScriptSetToMSBuildHookTargetMap => new Dictionary<string, TargetHookInfo>()
+        {
+            { "precompile",  new TargetHookInfo(true, "Build") },
+            { "postcompile", new TargetHookInfo(false, "Build") },
+            { "prepublish",  new TargetHookInfo(true, "Publish") },
+            { "postpublish", new TargetHookInfo(false, "Publish") }
+        };
+
+        private static Dictionary<string, string> ScriptVariableToMSBuildMap => new Dictionary<string, string>()
+        {
+            { "compile:TargetFramework", null },  // TODO: Need Short framework name in CSProj
+            { "compile:ResponseFile", null },     // Not migrated
+            { "compile:CompilerExitCode", null }, // Not migrated
+            { "compile:RuntimeOutputDir", null }, // Not migrated
+            { "compile:RuntimeIdentifier", null },// TODO: Need Rid in CSProj
```

Comment:
Actually in the cli this is only populated when you build for a rid, since we don't have that scenario anymore, this will be non-migrateable